### PR TITLE
Iss454 nr adjustments

### PR DESCRIPTION
**Pull request by jsphon**
https://github.com/liampauling/flumine/issues/454

### CODE REVIEW

File: flumine/markets/middleware.py [hunk 1]
```diff
@@ -99,7 +100,44 @@ def _process_runner_removal(
                         extra=order.info,
                     )
                 else:
-                    if (
+                    # TODO: "Where an SP lay bet in a win market has a maximum odds limit specified,..."
+                    if isinstance(order.order_type, MarketOnCloseOrder):
```
Comment (liampauling) [inline flumine/markets/middleware.py:104]: No problem with this but can we be consistant with the rest of the library and instead use:

```python
if order.order_type.ORDER_TYPE == OrderTypes.MARKET_ON_CLOSE:
```

File: flumine/markets/middleware.py [hunk 2]
```diff
@@ -99,7 +100,44 @@ def _process_runner_removal(
...
+                            logger.warning(
+                                "WIN MARKET_ON_CLOSE Order adjusted due to non runner {0}".format(
+                                    order.selection_id
+                                ),
+                                extra=order.info,
+                            )
+
+                        elif market.market_type in {"PLACE", "OTHER_PLACE"}:
+                            multiplier = (100 - removal_adjustment_factor) * 0.01
+
```
Comment (liampauling) [inline flumine/markets/middleware.py:130]: remove white space, don't think its needed

File: flumine/markets/middleware.py [hunk 3]
```diff
@@ -99,7 +100,44 @@ def _process_runner_removal(
...
+                            order.order_type.liability *= multiplier
+
```
Comment (liampauling) [inline flumine/markets/middleware.py:132]: remove white space, don't think its needed

**Review by liampauling — CHANGES_REQUESTED** [review]

File: flumine/markets/middleware.py (hunk 1 above)
Comment (liampauling) [reply flumine/markets/middleware.py:104]: Don't we need to filter this on LAY bets only?

`
For SP back bets, Betfair will at no point amend either the stake or the odds requested by a customer, despite any non-runners or withdrawals. However customers who have selected the SP limit option can reduce the minimum SP odds they are prepared to accept on a selection.
`

File: flumine/markets/middleware.py [hunk 4]
```diff
@@ -99,7 +99,40 @@ def _process_runner_removal(
...
+                                / (100 - runner_adjustment_factor)
+                            )
+                            order.order_type.liability *= multiplier
+                            logger.warning(
+                                "WIN MARKET_ON_CLOSE Order adjusted due to non runner {0}".format(
+                                    order.selection_id
+                                ),
+                                extra=order.info,
+                            )
+                        elif market.market_type in {"PLACE", "OTHER_PLACE"}:
```
Comment (liampauling) [inline flumine/markets/middleware.py:126]: Each way not included?

**Review by liampauling — CHANGES_REQUESTED** [review]

File: flumine/markets/middleware.py (hunk 4 above)
Comment (jsphon) [reply flumine/markets/middleware.py:126]: The rules page does not mentioned each way: https://www.betfair.com/aboutUs/Rules.and.Regulations/

Also, eyeballing a handful of eachway markets, I can't find any that offer BSP. Does BSP exist on each way markets?

**Review by liampauling — APPROVED** [review]

**Pull request merged**